src/main/plugins/org.dita.html5/xsl/task.xsl

```diff
@@ -131,9 +131,13 @@ See the accompanying LICENSE file for applicable license.
         <xsl:when test="*[contains(@class,' task/step ')] and not(*[contains(@class,' task/step ')][2])">
           <!-- Single step. Process any stepsection before the step (cannot appear after). -->
           <xsl:apply-templates select="*[contains(@class,' task/stepsection ')]"/>
-          <xsl:apply-templates select="*[contains(@class,' task/step ')]" mode="onestep">
-            <xsl:with-param name="step_expand" select="$step_expand"/>
-          </xsl:apply-templates>
+          <div>
+            <xsl:call-template name="commonattributes"/>
+            <xsl:call-template name="setid"/>
+            <xsl:apply-templates select="*[contains(@class,' task/step ')]" mode="onestep">
```

Comment:
The `step` template with `mode="onestep"` already calls `commonattributes` and `setidaname` so I think this will result in duplicate ID + classes?

Reply:
`onestep` calls `commonattributes` and `setidaname` for the `<step>` element, this calls `commonattributes` and `setid` for `<steps>` or `<steps-unordered>` element. So those are not the same.

Reply:
Ahh, of course, thanks.